src/workspacefactory/wfactory_controller.js

```diff
@@ -964,9 +1056,12 @@ WorkspaceFactoryController.prototype.importPreloadFromTree_ = function(tree) {
  * Clears the editing area completely, deleting all categories and all
  * blocks in the model and view and all pre-loaded blocks. Tied to the
  * "Clear" button.
+ *
+ * @param {boolean} conf Whether user should first confirm before clearing
+ *     workspace.
  */
-WorkspaceFactoryController.prototype.clearAll = function() {
-  if (!confirm('Are you sure you want to clear all of your work in Workspace' +
+WorkspaceFactoryController.prototype.clearAll = function(conf) {
+  if (conf && !confirm('Are you sure you want to clear all of your work in Workspace' +
```

Comment:
Adding the user confirmation means we should signal completion to the calling function, via callback or Promise.